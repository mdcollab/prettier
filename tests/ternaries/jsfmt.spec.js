run_spec(__dirname, null, ["typescript"]);
run_spec(__dirname, { tabWidth: 4 }, ["typescript"]);
