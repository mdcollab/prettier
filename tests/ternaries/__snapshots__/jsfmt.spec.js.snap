// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`binary.js 1`] = `
const funnelSnapshotCard = (report === MY_OVERVIEW &&
  !ReportGK.xar_metrics_active_capitol_v2) ||
  (report === COMPANY_OVERVIEW &&
    !ReportGK.xar_metrics_active_capitol_v2_company_metrics)
  ? <ReportMetricsFunnelSnapshotCard metrics={metrics} />
  : null;

room = room.map((row, rowIndex) => (
  row.map((col, colIndex) => (
    (rowIndex === 0 || colIndex === 0 || rowIndex === height || colIndex === width) ? 1 : 0
  ))
))
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const funnelSnapshotCard =
  (report === MY_OVERVIEW && !ReportGK.xar_metrics_active_capitol_v2) ||
  (report === COMPANY_OVERVIEW &&
    !ReportGK.xar_metrics_active_capitol_v2_company_metrics)
    ? <ReportMetricsFunnelSnapshotCard metrics={metrics} />
    : null;

room = room.map((row, rowIndex) =>
  row.map(
    (col, colIndex) =>
      rowIndex === 0 ||
      colIndex === 0 ||
      rowIndex === height ||
      colIndex === width
        ? 1
        : 0
  )
);

`;

exports[`binary.js 2`] = `
const funnelSnapshotCard = (report === MY_OVERVIEW &&
  !ReportGK.xar_metrics_active_capitol_v2) ||
  (report === COMPANY_OVERVIEW &&
    !ReportGK.xar_metrics_active_capitol_v2_company_metrics)
  ? <ReportMetricsFunnelSnapshotCard metrics={metrics} />
  : null;

room = room.map((row, rowIndex) => (
  row.map((col, colIndex) => (
    (rowIndex === 0 || colIndex === 0 || rowIndex === height || colIndex === width) ? 1 : 0
  ))
))
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const funnelSnapshotCard =
    (report === MY_OVERVIEW && !ReportGK.xar_metrics_active_capitol_v2) ||
    (report === COMPANY_OVERVIEW &&
        !ReportGK.xar_metrics_active_capitol_v2_company_metrics)
        ? <ReportMetricsFunnelSnapshotCard metrics={metrics} />
        : null;

room = room.map((row, rowIndex) =>
    row.map(
        (col, colIndex) =>
            rowIndex === 0 ||
            colIndex === 0 ||
            rowIndex === height ||
            colIndex === width
                ? 1
                : 0
    )
);

`;

exports[`indent.js 1`] = `
aaaaaaaaaaaaaaa ? bbbbbbbbbbbbbbbbbb : ccccccccccccccc ? ddddddddddddddd : eeeeeeeeeeeeeee ? fffffffffffffff : gggggggggggggggg
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
aaaaaaaaaaaaaaa
  ? bbbbbbbbbbbbbbbbbb
  : ccccccccccccccc
    ? ddddddddddddddd
    : eeeeeeeeeeeeeee ? fffffffffffffff : gggggggggggggggg;

`;

exports[`indent.js 2`] = `
aaaaaaaaaaaaaaa ? bbbbbbbbbbbbbbbbbb : ccccccccccccccc ? ddddddddddddddd : eeeeeeeeeeeeeee ? fffffffffffffff : gggggggggggggggg
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
aaaaaaaaaaaaaaa
    ? bbbbbbbbbbbbbbbbbb
    : ccccccccccccccc
      ? ddddddddddddddd
      : eeeeeeeeeeeeeee ? fffffffffffffff : gggggggggggggggg;

`;

exports[`parenthesis.js 1`] = `
debug ? this.state.isVisible ? "partially visible" : "hidden" : null;
debug ? this.state.isVisible && somethingComplex ? "partially visible" : "hidden" : null;

a => a ? () => {a} : () => {a}
a => a ? a : a
a => a ? aasdasdasdasdasdasdaaasdasdasdasdasdasdasdasdasdasdasdasdasdaaaaaa : a
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
debug ? (this.state.isVisible ? "partially visible" : "hidden") : null;
debug
  ? this.state.isVisible && somethingComplex ? "partially visible" : "hidden"
  : null;

a =>
  a
    ? () => {
        a;
      }
    : () => {
        a;
      };
a => (a ? a : a);
a =>
  a ? aasdasdasdasdasdasdaaasdasdasdasdasdasdasdasdasdasdasdasdasdaaaaaa : a;

`;

exports[`parenthesis.js 2`] = `
debug ? this.state.isVisible ? "partially visible" : "hidden" : null;
debug ? this.state.isVisible && somethingComplex ? "partially visible" : "hidden" : null;

a => a ? () => {a} : () => {a}
a => a ? a : a
a => a ? aasdasdasdasdasdasdaaasdasdasdasdasdasdasdasdasdasdasdasdasdaaaaaa : a
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
debug ? (this.state.isVisible ? "partially visible" : "hidden") : null;
debug
    ? this.state.isVisible && somethingComplex ? "partially visible" : "hidden"
    : null;

a =>
    a
        ? () => {
              a;
          }
        : () => {
              a;
          };
a => (a ? a : a);
a =>
    a ? aasdasdasdasdasdasdaaasdasdasdasdasdasdasdasdasdasdasdasdasdaaaaaa : a;

`;

exports[`test.js 1`] = `
const obj0 = conditionIsTruthy ? shortThing : otherShortThing

const obj1 = conditionIsTruthy ? { some: 'long', object: 'with', lots: 'of', stuff } : shortThing

const obj2 = conditionIsTruthy ? shortThing : { some: 'long', object: 'with', lots: 'of', stuff }

const obj3 = conditionIsTruthy ? { some: 'eeeeeeeeeeeeven looooooooooooooooooooooooooooooonger', object: 'with', lots: 'of', stuff } : shortThing

const obj4 = conditionIsTruthy ? shortThing : { some: 'eeeeeeeeeeeeven looooooooooooooooooooooooooooooonger', object: 'with', lots: 'of', stuff }

const obj5 = conditionIsTruthy ? { some: 'long', object: 'with', lots: 'of', stuff } : { some: 'eeeeeeeeeeeeven looooooooooooooooooooooooooooooonger', object: 'with', lots: 'of', stuff }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const obj0 = conditionIsTruthy ? shortThing : otherShortThing;

const obj1 = conditionIsTruthy
  ? { some: "long", object: "with", lots: "of", stuff }
  : shortThing;

const obj2 = conditionIsTruthy
  ? shortThing
  : { some: "long", object: "with", lots: "of", stuff };

const obj3 = conditionIsTruthy
  ? {
      some: "eeeeeeeeeeeeven looooooooooooooooooooooooooooooonger",
      object: "with",
      lots: "of",
      stuff
    }
  : shortThing;

const obj4 = conditionIsTruthy
  ? shortThing
  : {
      some: "eeeeeeeeeeeeven looooooooooooooooooooooooooooooonger",
      object: "with",
      lots: "of",
      stuff
    };

const obj5 = conditionIsTruthy
  ? { some: "long", object: "with", lots: "of", stuff }
  : {
      some: "eeeeeeeeeeeeven looooooooooooooooooooooooooooooonger",
      object: "with",
      lots: "of",
      stuff
    };

`;

exports[`test.js 2`] = `
const obj0 = conditionIsTruthy ? shortThing : otherShortThing

const obj1 = conditionIsTruthy ? { some: 'long', object: 'with', lots: 'of', stuff } : shortThing

const obj2 = conditionIsTruthy ? shortThing : { some: 'long', object: 'with', lots: 'of', stuff }

const obj3 = conditionIsTruthy ? { some: 'eeeeeeeeeeeeven looooooooooooooooooooooooooooooonger', object: 'with', lots: 'of', stuff } : shortThing

const obj4 = conditionIsTruthy ? shortThing : { some: 'eeeeeeeeeeeeven looooooooooooooooooooooooooooooonger', object: 'with', lots: 'of', stuff }

const obj5 = conditionIsTruthy ? { some: 'long', object: 'with', lots: 'of', stuff } : { some: 'eeeeeeeeeeeeven looooooooooooooooooooooooooooooonger', object: 'with', lots: 'of', stuff }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const obj0 = conditionIsTruthy ? shortThing : otherShortThing;

const obj1 = conditionIsTruthy
    ? { some: "long", object: "with", lots: "of", stuff }
    : shortThing;

const obj2 = conditionIsTruthy
    ? shortThing
    : { some: "long", object: "with", lots: "of", stuff };

const obj3 = conditionIsTruthy
    ? {
          some: "eeeeeeeeeeeeven looooooooooooooooooooooooooooooonger",
          object: "with",
          lots: "of",
          stuff
      }
    : shortThing;

const obj4 = conditionIsTruthy
    ? shortThing
    : {
          some: "eeeeeeeeeeeeven looooooooooooooooooooooooooooooonger",
          object: "with",
          lots: "of",
          stuff
      };

const obj5 = conditionIsTruthy
    ? { some: "long", object: "with", lots: "of", stuff }
    : {
          some: "eeeeeeeeeeeeven looooooooooooooooooooooooooooooonger",
          object: "with",
          lots: "of",
          stuff
      };

`;
